File: 0055-jump-game/0055-jump-game.c

bool canJump(int* nums, int numsSize) {
    for (int i = 0; i < numsSize; i++) {
        if (i > nums[0]) {
            return false;  
        }
        if (i + nums[i] > nums[0]) {
            nums[0] = i + nums[i]; 
        }
    }
    return true;
}

File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int EndIndex = 0;
        for(int i = 0; i < nums.length; i++){
            if(i > EndIndex) return false;
            EndIndex = Math.max(EndIndex,nums[i]+i);
        }
        return true;
    }
}